src/components/ZoomDropdown/ZoomDropdown.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import DropdownItem from '../DropdownItem';
+
+import core from 'core';
+import getOverlayPositionBasedOn from 'helpers/getOverlayPositionBasedOn';
+import { zoomTo } from 'helpers/zoom';
+import actions from 'actions';
+import selectors from 'selectors';
+
+import './ZoomDropdown.scss';
+
+class ZoomDropdown extends React.PureComponent {
+  static propTypes = {
+    isOpen: PropTypes.bool,
+    closeElements: PropTypes.func.isRequired
+  }
+
+  constructor() {
+    super();
+    this.dropdown = React.createRef();
+    this.state = {
+      left: 0,
+      right: 'auto'
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', this.handleWindowResize);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (!prevProps.isOpen && this.props.isOpen) {
+      this.props.closeElements([ 'viewControlsOverlay', 'toolsOverlay', 'searchOverlay', 'menuOverlay', 'toolStylePopup' ]);
+      this.setState(getOverlayPositionBasedOn('zoomDropdownButton', this.dropdown),()=>{
+        this.setState({ left: this.state.left - 20 });
+      });
```

Comment:
Same here, instead of using setState twice, I think we can doing something like:
```javascript
const { left, right } = getOverlayPositionBasedOn('zoomDropdownButton', this.dropdown);
this.setState({ 
  left: left - 20,
  right
});
```
This way, this component will only re-render once.

Also I see `getOverlayPositionBasedOn` has a third parameter now, maybe we can use `getOverlayPositionBasedOn('zoomDropdownButton', this.dropdown, 'center')`?

Reply:
The reason why `center` doesn't work on `getOverlayPositionBasedOn()` is because we are using custom element. Because of this, the reference becomes the textarea part of the button, instead of the whole button.